src/Compilers/CSharp/Portable/Compilation/MethodBodySemanticModel.cs

```diff
@@ -30,13 +28,13 @@ internal static MethodBodySemanticModel Create(CSharpCompilation compilation, Me
         }
 
         /// <summary>
-        /// Creates a SemanticModel for an ArrowExpressionClause, which includes
-        /// an ExecutableCodeBinder and a ScopedExpressionBinder.
+        /// Creates a SemanticModel for a GlobalStatementSyntax.
         /// </summary>
-        internal static MethodBodySemanticModel Create(CSharpCompilation compilation, MethodSymbol owner, Binder rootBinder, ArrowExpressionClauseSyntax syntax)
+        internal static MethodBodySemanticModel Create(CSharpCompilation compilation, SynthesizedInteractiveInitializerMethod scriptInitializer, Binder rootBinder, GlobalStatementSyntax syntax)
         {
-            Binder binder = new ExecutableCodeBinder(syntax, owner, rootBinder);
-            return new MethodBodySemanticModel(compilation, owner, binder, syntax);
+            var compilationUnit = (CompilationUnitSyntax)syntax.Parent;
+            var binder = new ExecutableCodeBinder(compilationUnit, scriptInitializer, rootBinder);
+            return new MethodBodySemanticModel(compilation, scriptInitializer, binder.GetBinder(compilationUnit), syntax);
```

Comment:
Are we creating MethodBodySemanticModel for each global statement? Is this the right thing to do? Shouldn't all global statements share the same scope, local symbols, etc.?